tools/pythonpkg/src/vector_conversion.cpp

```diff
@@ -86,13 +86,24 @@ bool ValueIsNull(double value) {
 }
 
 template <class T>
-void ScanPandasFpColumn(T *src_ptr, idx_t count, idx_t offset, Vector &out) {
-	FlatVector::SetData(out, (data_ptr_t)(src_ptr + offset));
-	auto tgt_ptr = FlatVector::GetData<T>(out);
+void ScanPandasFpColumn(T *src_ptr, idx_t stride, idx_t count, idx_t offset, Vector &out) {
 	auto &mask = FlatVector::Validity(out);
-	for (idx_t i = 0; i < count; i++) {
-		if (Value::IsNan<T>(tgt_ptr[i])) {
-			mask.SetInvalid(i);
+	if (stride == sizeof(T)) {
+		FlatVector::SetData(out, (data_ptr_t)(src_ptr + offset));
```

Comment:
Might not be worth it to separate these two cases, as we need to loop over the data regardless to check for NaN